gui/default/syncthing/core/syncthingController.js

```diff
@@ -272,6 +272,11 @@ angular.module('syncthing.core')
             $scope.pendingFolders[arg.data.folder] = pendingFolder;
         });
 
+        $scope.$on(Events.FOLDER_OFFER_CANCELLED, function (event, arg) {
```

Comment:
I don't think we should have a cancel event.

Effectively what we care about, is when the folder list is updated.
So far we used to rely on folder_rejected event/config_saved, but perhaps it now should be pending_folders_changed, or something.
If you really care what was added what was removed the "data" part of the event could contain that.

Reply:
So far there was only `FolderRejected` which contained all details so that the GUI updated its list in-place without querying the endpoint (formerly config, now cluster) again. I didn't want to make a breaking event API change, and reusing the event with that name would be confusing anyway.

However, no-longer-pending folders should happen very rarely and in my opinion querying the endpoint again and replacing the whole list is justified (as is done on config changes).

So we can just rename the event to `PendingFoldersChanged` or even more generic `ClusterPendingChanged`. But I don't think removing `FolderRejected` should be touched or replaced. Not sure if you were suggesting that.

Reply:
We can use the `ClusterPendingChanged` event also after removing stuff in `cleanPending()`.  It's currently handled through the `ConfigSaved` event, but that coupling leads to unnecessary API queries when nothing actually changed.

Reply:
Why do we have to wedge in the term cluster here?

There are some pending folders and pending devices, how does adding cluster suddenly give this more meaning?

Syncthing is already heavy on new terminology, so I think we should strive to minimise new magical words that we expose to the users/external developers, as they cause confusion.

Reply:
The term ClusterPending refers to the API endpoints `/rest/cluster/pending/*`.  Its intended meaning is "refreshing info from those endpoints should be done now, as it has changed in an unspecified way".

Cluster is used in the docs IIRC and to me is related to the structure which devices and shared folders are connected to each other.

Reply:
I agree in principle. However only "pending" feels very obscure too - what is "pending"? And I can't easily come up with a convincing alternative for the meaning of "linked devices and folders" or "introducing/altering device/folder pairings". Maybe "pairing"? Maybe that's more of a common/comprehensible word than "cluster" and could work as `PendingPairingChanged` and `/rest/pairing/pending/folders`. I am not terribly convinced though.

Reply:
I need to get my "candidate folders / devices" branch cleaned up and ready for PR. Because that's always in the back of my head when searching for the right terminology and it will also live under `/rest/cluster/candidate/*`. The endpoint name btw. is now already in the docs, no?

@imsodin are you unhappy with the endpoint naming or just the event name? It's meant to spell out explicitly that exactly this endpoint (subtree) needs refreshing.

Reply:
The candidate thing just plays into the same: It could also be `/rest/pairing/candidate/...`, or just `/rest/candidate/...`. All my the already mentioned points apply to that as well. And I am still not super happy with either "cluster" or "pairing". Then again, maybe we are bikeshedding here and should just accept "cluster" as a fact. I don't care much either way, I just tried to express some ideas to hopefully advance the discussion towards a solution.

They (endpoints and events) definitely need to be consistent. However the new `/rest/cluster/...` stuff hasn't been released, so now is the time to change it if we want to.

Reply:
I slightly prefer cluster over pairing, but could live with either. Since cluster is in fact mentioned several times in the documentation, I thought it was common terminology on the developer front.

Reply:
The folder and device is pending sharing?

I don't like candidate or pairing as it's yet a new word.

If we converge on something else, we should definitely update end rest urls.

Reply:
Candidate stuff is (will be) in fact new functionality, so a new term seems justified. It's closely related to pending though, both derived from the ClusterConfig message contents. Guess that's where I got the "cluster" from...

I agree this is turning into bikeshedding though, so will try to focus on agreed changes until someone comes up with a better term that could lead to consensus.